0-module-and-module-source.emu

```diff
@@ -548,8 +550,11 @@ location: https://tc39.es/proposal-compartments/
                 1. Else, if Type(_importMeta_) is not Object, then
                   1. Throw a *TypeError* exception.
                 1. Perform ? RequireInternalSlot(_moduleSource_, [[ModuleSource]]).
+                1. Let _evalRealm_ be the current Realm Record.
+                1. NOTE: Module Instances created from this constructor must use a valid source.
+                1. Perform ? HostEnsureCanCompileStrings(_evalRealm_).
                 1. Let _O_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%Module.prototype%"*, « [[Module]], [[ModuleSourceInstance]], [[ImportHook]] »).
-                1. Let _moduleRecord_ to ! CreateModuleRecord(_moduleSource_.[[ModuleSource]]).
+                1. Let _moduleRecord_ to ! CreateModuleRecord(_evalRealm_, _moduleSource_.[[ModuleSource]]).
```

Comment:
reusing the `_evalRealm_` here, but also it makes sense that when a module record is created, it can be associated to a realm, and the caller of the abstract operation to be responsible for providing such realm. This opens the door for more advanced use-cases like transferring instances across realms.